app/javascript/app/components/table/table-component.jsx

```diff
@@ -58,7 +59,9 @@ function TableComponent(props) {
 
   if (!data || !data.length) return null;
   const hasColumnSelectedOptions = hasColumnSelect && columnsOptions;
-  const activeColumnNames = activeColumns.map(c => c.value);
+  const activeColumnNames = activeColumns
+    .map(c => c.value)
+    .filter(c => !hiddenColumns.includes(c));
```

Comment:
I'm not 100% sure about this. 

I mostly wanted to avoid the `activeColumns` implementation altogether to prevent bugs, as it seems to rely on the column labels to end with the `NotShow` string. Because it affects other modules and due to time constraints I worked around this, though I believe it'd be ideal to refactor it in the future.

Reply:
Actually I'm not sure how this is working as what should include the hiddenColumns is the c.value not the c

Reply:
Yes, maybe we can filter that column before it arrives to the table-component in the future